### ISSUE TITLE
regression: non-object mode concatenates last string chunk with `,`

### ISSUE THREAD

**Issue by derhuerst** [Floby/node-stream-sink#6]
```js
'use strict'

const fs = require('fs')
const sink = require('stream-sink')

fs.createReadStream('/etc/hosts', {encoding: 'base64'})
.pipe(sink())
.then(buf => console.log(buf))
.catch(console.error)
```

With `stream-sink@2.2.1`:

```
IyMKIyBIb3N0IERhdGFiYXNlCiMKIyBsb2NhbGhvc3QgaXMgdXNlZCB0byBjb25maWd1cmUgdGhlIGxvb3BiYWNrIGludGVyZmFjZQojIHdoZW4gdGhlIHN5c3RlbSBpcyBib290aW5nLiAgRG8gbm90IGNoYW5nZSB0aGlzIGVudHJ5LgojIwoxMjcuMC4wLjEJbG9jYWxob3N0CjI1NS4yNTUuMjU1LjI1NQlicm9hZGNhc3Rob3N0Cjo6MSAgICAgICAgICAgICBsb2NhbGhvc3QgCgowLjAuMC4wICAgICAgICAgZ29sZW0uZGUKMC4wLjAuMCAgICAgICAgIHd3dy5nb2xlbS5k,ZQo=
```

With `stream-sink@2.1.1`:

```
IyMKIyBIb3N0IERhdGFiYXNlCiMKIyBsb2NhbGhvc3QgaXMgdXNlZCB0byBjb25maWd1cmUgdGhlIGxvb3BiYWNrIGludGVyZmFjZQojIHdoZW4gdGhlIHN5c3RlbSBpcyBib290aW5nLiAgRG8gbm90IGNoYW5nZSB0aGlzIGVudHJ5LgojIwoxMjcuMC4wLjEJbG9jYWxob3N0CjI1NS4yNTUuMjU1LjI1NQlicm9hZGNhc3Rob3N0Cjo6MSAgICAgICAgICAgICBsb2NhbGhvc3QgCgowLjAuMC4wICAgICAgICAgZ29sZW0uZGUKMC4wLjAuMCAgICAgICAgIHd3dy5nb2xlbS5kZQo=
```

macOS High Sierra, Node v9.7.1

**Comment by derhuerst** [comment]
@Floby ping (:

**Comment by derhuerst** [comment]
This should be fixed as of e9c43bc9f78be1a9e2e40db203a033297d65c0f1 and `stream-sink@2.2.2`. Thanks!

**Comment by Floby** [comment]
Yeah sorry about that, seems its dependent on your version of node.
Glad to see someone using this though. Thanks! :)
